spec/lib/api/login-spec.js

```diff
@@ -0,0 +1,377 @@
+                    expect(res.body.message).to.equal('Missing credentials');
+                });
+        });
+
+        it('should fail with no username parameter in query string', function() {
+            return helper.request('https://localhost:8443')
+                .get('/login?password=admin123')
+                .send()
+                .expect(BAD_REQUEST_STATUS)
+                .expect(function(res) {
+                    expect(res.body.message).to.be.a('string');
+                    expect(res.body.message).to.equal('Missing credentials');
+                });
+        });
+
+        it('should fail with no password parameter in query string', function() {
+            return helper.request('https://localhost:8443')
+                .get('/login?username=admin')
+                .send()
+                .expect(BAD_REQUEST_STATUS)
+                .expect(function(res) {
+                    expect(res.body.message).to.be.a('string');
+                    expect(res.body.message).to.equal('Missing credentials');
+                });
+        });
+
+        //passport-local middleware we choose does not support authentication
+        // with credential in the header. Following test will fail if auth header
+        // is supported in the future, thus people will get alerted.
+        it('should fail with credential in request header', function() {
+            return helper.request('https://localhost:8443')
+                .post('/login')
+                .set('username', 'admin')
+                .set('password', 'admin123')
+                .expect(BAD_REQUEST_STATUS)
+                .expect(function(res) {
+                    expect(res.body.message).to.be.a('string');
+                    expect(res.body.message).to.equal('Missing credentials');
+                });
+        });
+
+        it('should fail no credential at all - https', function() {
+            return helper.request('https://localhost:8443')
+                .post('/login')
+                .expect(BAD_REQUEST_STATUS)
+                .expect(function(res) {
+                    expect(res.body.message).to.be.a('string');
+                    expect(res.body.message).to.equal('Missing credentials');
+                });
+        });
+
+        after('stop server, restore mock and configure',function () {
+            cleanUp();
+        });
+    });
+
+    describe('test with authentication enabled', function () {
+        before('start HTTPs server', function () {
+            this.timeout(5000);
+            var endpointHttp = {
+                "address": "0.0.0.0",
+                "port": 8089,
+                "httpsEnabled": false,
+                "authEnabled": true,
+                "routers": "northbound-api-router"
+            };
+            startServer(endpointHttp);
+        });
+
+        //give a shoot on http instead of https.
+        it('should success auth with http instead of https', function() {
```

Comment:
@zyoung51 we do not recommend user to do that but the code did not restrict them from doing it. This test case is to make sure we will not crash even user choose to do auth over http.